### PR TITLE
upload one chunk

### PR DESCRIPTION
**Pull request by steeeee99**
# Description

Added a new endpoint to upload and store just one chunk of text.

Related to issue #716 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by zAlweNy26** [comment]
what is the benefit of this instead of just create a file on-the-fly and send it through the existing endpoint?

**Comment by pieroit** [comment]
> What is the benefit of this instead of just creating a file on the fly and sending it through the existing endpoint?

I opened an issue for this feat
Just text plus metadata, so any other parsing and splitting can be done client side

**Comment by pieroit** [comment]
@steeeee99 thanks, reviewing soon

File: core/cat/routes/upload.py [hunk 1]
```diff
@@ -54,6 +54,28 @@ async def upload_file(
...
 
+# receive chunk via http endpoint
+@router.post("/chunk")
+async def upload_chunk(
+    request: Request,
+    background_tasks: BackgroundTasks,
+    chunk: str = Body(
+        description="A chunk to upload in the cat memory",
+        examples={"Example of chunk with length greater than 20"},
+        min_length=20
```
Comment (nicola-corbellini) [inline core/cat/routes/upload.py:65]: is there any particular reason for the choice of 20?

Comment (steeeee99) [reply core/cat/routes/upload.py:65]: no particular reason, what value is best to have? 10 like [here](https://github.com/cheshire-cat-ai/core/blob/4c9fff02470ecd406a346652d035aa98ada71d58/core/cat/rabbit_hole.py#L418C9-L419C1) ?

**Comment by pieroit** [comment]
> > > What is the benefit of this instead of just creating a file on the fly and sending it through the existing endpoint?
> > 
> > 
> > I opened an issue for this feat Just text plus metadata, so any other parsing and splitting can be done client side
> 
> Hey @pieroit, I'm uploading only a string, we need also metadata ? with which fields?

Hi @steeeee99, 
thanks for this contribution.

`metadata` should be a dictionary, containing anything (because metadata is custom)
[Here](https://github.com/cheshire-cat-ai/core/blob/4c9fff02470ecd406a346652d035aa98ada71d58/core/cat/rabbit_hole.py#L337) is how we fill up default metadata.

**Comment by pieroit** [comment]
@steeeee99 any progress on this?
Please make the PR branching off `develop` and towards `develop`, main is just for releases

**Comment by pieroit** [comment]
Closing as this feat should be made according to #764 and not related to the rabbit hole

**Pull request closed without merging**